cpp/src/community/k_truss_impl.cuh

```diff
@@ -41,100 +43,36 @@
-      [query_vertices,
-       num_edges       = sorted_vertices.size(),
-       sorted_vertices = sorted_vertices.begin()] __device__(size_t idx) {
-        auto itr_lower = thrust::lower_bound(
-          thrust::seq, sorted_vertices, sorted_vertices + num_edges, query_vertices[idx]);
-
-        auto itr_upper = thrust::upper_bound(
-          thrust::seq, itr_lower, sorted_vertices + num_edges, query_vertices[idx]);
-        vertex_t dist = thrust::distance(itr_lower, itr_upper);
-
-        return dist;
-      }));
-
-  thrust::exclusive_scan(handle.get_thrust_policy(),
-                         count_first,
-                         count_first + query_vertices.size() + 1,
-                         prefix_sum.begin());
-
-  return prefix_sum;
-}
-
-template <typename vertex_t, typename edge_t, typename EdgeIterator>
+template <typename vertex_t, typename edge_t, typename EdgeIterator, bool is_q_r_edge>
 edge_t remove_overcompensating_edges(raft::handle_t const& handle,
                                      size_t buffer_size,
                                      EdgeIterator potential_closing_or_incoming_edges,
                                      EdgeIterator incoming_or_potential_closing_edges,
-                                     raft::device_span<vertex_t const> invalid_edgelist_srcs,
-                                     raft::device_span<vertex_t const> invalid_edgelist_dsts)
+                                     raft::device_span<vertex_t const> weak_edgelist_first_srcs,
+                                     raft::device_span<vertex_t const> weak_edgelist_first_dsts)
 {
   // To avoid over-compensating, check whether the 'potential_closing_edges'
-  // are within the invalid edges. If yes, the was already unrolled
+  // are within the weak edges. If yes, the was already unrolled
   auto edges_not_overcomp = thrust::remove_if(
     handle.get_thrust_policy(),
     thrust::make_zip_iterator(potential_closing_or_incoming_edges,
                               incoming_or_potential_closing_edges),
     thrust::make_zip_iterator(potential_closing_or_incoming_edges + buffer_size,
                               incoming_or_potential_closing_edges + buffer_size),
-    [num_invalid_edges = invalid_edgelist_dsts.size(),
-     invalid_first =
-       thrust::make_zip_iterator(invalid_edgelist_dsts.begin(), invalid_edgelist_srcs.begin()),
-     invalid_last = thrust::make_zip_iterator(invalid_edgelist_dsts.end(),
-                                              invalid_edgelist_srcs.end())] __device__(auto e) {
+    [num_weak_edges = weak_edgelist_first_dsts.size(),
+     weak_first =
+       thrust::make_zip_iterator(weak_edgelist_first_srcs.begin(), weak_edgelist_first_dsts.begin()),
+     weak_last = thrust::make_zip_iterator(weak_edgelist_first_srcs.end(),
+                                              weak_edgelist_first_dsts.end())] __device__(auto e) {
       auto potential_edge = thrust::get<0>(e);
```

Comment:
We are not using `thrust::get<1>(e)` but we still need to pass it because we are removing the pair (`potential_closing_or_incoming_edges`, `incoming_or_potential_closing_edges`), not just `potential_closing_or_incoming_edges`